src/OrchardCore/OrchardCore.Mvc.Core/LocationExpander/ModularViewLocationExpanderProvider.cs

```diff
@@ -54,7 +71,62 @@ IEnumerable<string> PageViewLocations()
 
             var extensionViewsPath = '/' + extension.SubPath + "/Views";
             result.Add(extensionViewsPath + "/{1}/{0}" + RazorViewEngine.ViewExtension);
-            result.Add(extensionViewsPath + "/Shared/{0}" + RazorViewEngine.ViewExtension);
+
+            if (!context.ViewName.StartsWith("Components/", StringComparison.Ordinal))
```

Comment:
Is that a new custom folder? Or already something from MVC?

Reply:
From MVC as [defined here](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponents/ViewViewComponentResult.cs#L23).